### DISCUSSION TITLE
Pass data to MSW handlers, like the create fake data functions.

### DISCUSSION THREAD

**Discussion by Samuel-Morgan-Tyghe** [kubb-labs/kubb#1062] [Ideas]
So theres alot of uses for passing in hardcoded values to our fake data, but to utilize this we then have to pass it into a handler. 

why not pass data directly into the handler into the handler?

**Comment by Samuel-Morgan-Tyghe** [comment]
I need an easy way to extend kubb msw

looking at the handlers they are not functions, so this may be difficult to intergrate. 

currently i have to manually add a new handler, pass in generateResponse('hardcoded data')  and its quite tedius. 

but what i can do to shorten this is using generatedHandler.info.path for the path of the new handler

Bottom line wouldnt it be cleaner to be able to reference the handler and modify it than to replicate it.

**Reply by stijnvanhulle** [reply]
How do you see it being generated as in do you have an example of what your perfect code should look like? Then I can have a look to do some changes based on that :)

**Reply by Samuel-Morgan-Tyghe** [reply]
```
import { http } from "msw";
import { createGetProductSummaryMutationResponse } from "../../mocks/retail/createGetProductSummary";


export const getProductSummaryHandler = (extendedData) => http.post("*/v2/products/summary", function handler() {
    return new Response(JSON.stringify(createGetProductSummaryMutationResponse(extendedData)), {
        headers: {
            "Content-Type": "application/json",
        },
    });
});
```


```
export function createGetProductSummaryMutationResponse(extendedData): NonNullable<GetProductSummaryMutationResponse> {
    faker.seed([100]);
    return createProductSummary(extendedData);
}
```

**Reply by oskar-dragon** [reply]
@stijnvanhulle Do you happen to have any updates on this? This would make testing much easier. The way Kubb currently generates MSW handlers is not very useful. Orval allows to do it so it would be worth taking some inspiration from them.

https://orval.dev/guides/msw

**Reply by stijnvanhulle** [reply]
@oskar-dragon Created an issue so that could be tackled later on: https://github.com/kubb-labs/kubb/issues/1114.
